src/NServiceBus.Transport.RabbitMQ/Connection/ChannelProvider.cs

```diff
@@ -21,36 +21,45 @@ public ChannelProvider(ConnectionFactory connectionFactory, TimeSpan retryDelay,
 
         public void CreateConnection()
         {
+            if (connection is not null)
+            {
+                connection.ConnectionShutdown -= Connection_ConnectionShutdown;
+                connection.Dispose();
+                connection = null;
+            }
             connection = connectionFactory.CreatePublishConnection();
             connection.ConnectionShutdown += Connection_ConnectionShutdown;
         }
 
+        Task reconnectTask = Task.CompletedTask;
+
         void Connection_ConnectionShutdown(object sender, ShutdownEventArgs e)
         {
             if (e.Initiator != ShutdownInitiator.Application)
             {
                 var connection = (IConnection)sender;
 
-                // Task.Run() so the call returns immediately instead of waiting for the first await or return down the call stack
-                _ = Task.Run(() => ReconnectSwallowingExceptions(connection.ClientProvidedName), CancellationToken.None);
```

Comment:
The `Task.Run` is intentional here and needs to stay. We need to ensure that the reconnection code is happening on a different thread than the internal thread from the client that event is called from.

Reply:
That happens anyway because the first thing the code does is asynchronously yielding in the delay.

Reply:
And then when the next change comes later that removes that, we've then also lost the context for why it needs to be on another thread. Still disagree with this change.

Reply:
Reverted the Task.Run and clarified the code comment